### PR TITLE
[feat] Add LTX-2 and LTX-2.3 fine-tuning to the modular trainer

### PR DESCRIPTION
**Pull request by SolitaryThinker**
## Summary

- add an LTX2Model plugin for standard-diffusion fine-tuning in `fastvideo/train`
- support LTX-2.0 and LTX-2.3 checkpoint and text-embedding differences
- add overfit preprocessing/configs plus model-load, CPU contract, single-step,
  and nightly end-to-end coverage

## Validation

- 4x GB200, 300-step overfit: LTX-2.0 loss 0.51 -> 0.012 and validation
  MS-SSIM 0.15 -> 0.96
- 4x GB200, 300-step overfit: LTX-2.3 loss -> 0.0085 and validation
  MS-SSIM 0.127 -> 0.957
- parameterized nightly end-to-end test passed for both checkpoints
  (0.964 / 0.957)
- pre-commit passed on the rebased diff; all four training configs parse on
  current main

## Notes

- video-only fine-tuning freezes unused audio/cross-modal parameters; joint
  audio training is rejected until its data and loss path is implemented
- the high-memory checkpoint tests skip on the 48 GB L40S lane, while the new
  CPU contract tests exercise wrapper math, context restoration, freezing, and
  both checkpoint variants in required PR CI
- this extends the modular `fastvideo/train` stack; it does not migrate or
  replace the existing legacy LTX-2 trainer

### CODE REVIEW

File: fastvideo/pipelines/preprocess/preprocess_ltx2_overfit.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+def load_video(path: str, num_frames: int, target_fps: float, height: int,
+               width: int) -> tuple[torch.Tensor, np.ndarray]:
+    """Load a video as [1, C, T, H, W] in [-1, 1], resampled to target_fps.
+
+    Also returns the uint8 RGB frames [T, H, W, C] for reference-video export.
+    """
+    with av.open(path) as container:
+        stream = container.streams.video[0]
+        native_fps = float(stream.average_rate or target_fps)
```
[the remark this answers is not in the document]
Comment (SolitaryThinker) [reply fastvideo/pipelines/preprocess/preprocess_ltx2_overfit.py:78]: Fixed in 935bb437d. load_video now rejects containers with no video stream using a descriptive RuntimeError before indexing, with a CPU regression using an audio-only WAV.

File: fastvideo/train/models/ltx2/ltx2.py [hunk 2]
```diff
@@ -0,0 +1,562 @@
...
+        if isinstance(denoised, tuple):
+            denoised = denoised[0]
+
+        # The wrapper returns the denoised x0 prediction
+        # (sample - velocity * sigma). Convert back to velocity so the
+        # framework's default target ``noise - clean`` applies. Compute
+        # in fp32: sigma can be ~1e-3 and bf16 division would amplify
+        # quantization error.
+        sigmas = batch.sigmas.float()
+        velocity = (noisy_bcthw.float() - denoised.float()) / sigmas
```
[the remark this answers is not in the document]
Comment (SolitaryThinker) [reply fastvideo/train/models/ltx2/ltx2.py:351]: Not applying the clamp. _sample_ltx2_sigmas guarantees sigma >= 1e-3: low values are reflected around epsilon and uniform samples start at epsilon; the same tensor is passed to predict_noise. At sigma 0 velocity is mathematically undefined, so clamping would hide a broken batch invariant and manufacture a value. The official LTX conversion rejects zero rather than clamping.
